File: problems/1805-number-of-different-integers-in-a-string/solution.py
class Solution:
    def numDifferentIntegers(self, word: str) -> int:
        nums = set()
        current_num = ""
        for char in word:
            if char.isdigit():
                current_num += char
            else:
                if current_num:
                    nums.add(int(current_num))
                    current_num = ""
        if current_num:
            nums.add(int(current_num))
        return len(nums)

File: problems/1805-number-of-different-integers-in-a-string/solution.cpp
class Solution {
public:
    int numDifferentIntegers(string word) {
        unordered_set<string> s;
        string num = "";
        for (char c : word) {
            if (isdigit(c)) {
                num += c;
            } else {
                if (!num.empty()) {
                    int i = 0;
                    while (i < num.length() - 1 && num[i] == '0') {
                        i++;
                    }
                    s.insert(num.substr(i));
                    num = "";
                }
            }
        }
        if (!num.empty()) {
            int i = 0;
            while (i < num.length() - 1 && num[i] == '0') {
                i++;
            }
            s.insert(num.substr(i));
        }
        return s.size();
    }
};